wp_mobile_cache/src/repository/term_relationships.rs

```diff
@@ -0,0 +1,483 @@
+    ///
+    /// This approach is observer-friendly: unchanged terms generate no DB events.
+    /// Only actual changes (new terms added, old terms removed) generate INSERT/DELETE events.
+    pub fn sync_terms_for_object(
+        &self,
+        executor: &impl QueryExecutor,
+        site: &DbSite,
+        object_id: RowId,
+        taxonomy_type: &TaxonomyType,
+        new_term_ids: &[TermId],
+    ) -> Result<(), SqliteDbError> {
+        // 1. Get existing term IDs
+        let existing_terms = self.get_terms_for_object(executor, site, object_id, taxonomy_type)?;
+
+        // 2. Calculate diff (using Vec-based filtering since TermId may not impl Hash)
+        let to_delete: Vec<_> = existing_terms
+            .iter()
+            .filter(|existing| !new_term_ids.contains(existing))
+            .copied()
+            .collect();
+
+        let to_insert: Vec<_> = new_term_ids
+            .iter()
+            .filter(|new_id| !existing_terms.contains(new_id))
+            .copied()
+            .collect();
+
+        // 3. Delete removed terms (only the ones being removed)
+        if !to_delete.is_empty() {
+            self.delete_terms(executor, site, object_id, taxonomy_type, &to_delete)?;
+        }
+
+        // 4. Insert new terms (only the ones being added)
+        if !to_insert.is_empty() {
+            self.insert_terms(executor, site, object_id, taxonomy_type, &to_insert)?;
+        }
+
+        // Unchanged terms: no DB operations = no observer events ✅
+        Ok(())
+    }
+
+    /// Delete specific terms for an object.
+    fn delete_terms(
+        &self,
+        executor: &impl QueryExecutor,
+        site: &DbSite,
+        object_id: RowId,
+        taxonomy_type: &TaxonomyType,
+        term_ids: &[TermId],
+    ) -> Result<(), SqliteDbError> {
+        if term_ids.is_empty() {
+            return Ok(());
+        }
+
+        // Build placeholders for IN clause
+        let placeholders: Vec<_> = (0..term_ids.len()).map(|_| "?").collect();
+        let sql = format!(
+            "DELETE FROM term_relationships WHERE db_site_id = ? AND object_id = ? AND taxonomy_type = ? AND term_id IN ({})",
+            placeholders.join(", ")
+        );
+
+        // Build params: [site_id, object_id, taxonomy_type, term_id1, term_id2, ...]
+        let mut params: Vec<Box<dyn rusqlite::ToSql>> = vec![
+            Box::new(site.row_id),
+            Box::new(object_id),
+            Box::new(taxonomy_type.to_string()),
+        ];
+        for term_id in term_ids {
+            params.push(Box::new(term_id.0));
+        }
+
+        let params_refs: Vec<_> = params.iter().map(|p| p.as_ref()).collect();
+        executor.execute(&sql, params_refs.as_slice())?;
+        Ok(())
+    }
+
+    /// Insert new terms for an object.
+    fn insert_terms(
+        &self,
+        executor: &impl QueryExecutor,
+        site: &DbSite,
+        object_id: RowId,
+        taxonomy_type: &TaxonomyType,
+        term_ids: &[TermId],
```

Comment:
We probably don't need to insert term_ids directly and should use the sync one instead.